src/lib/stores/orderStore.ts

```diff
@@ -39,6 +56,76 @@ export default {
   }
 };
 
+const attachCustomizationsToOrderItem = async (
+  orderItemId: RecordIdString,
+  customizations: CustomizationValue[]
+) => {
+  const itemCustomizationIds = await createItemCustomizations(
+    R.groupBy(customizations, (customization) => customization.belongsTo)
+  );
+
+  if (!R.isEmpty(itemCustomizationIds)) {
+    await pb.collection(Collections.OrderItem).update(orderItemId, {
+      customization: itemCustomizationIds
+    });
+  }
+};
+
+const createItemCustomizations = async (
+  customizationsByKey: Record<string, CustomizationValue[]>
+) => {
+  const entries = Object.entries(customizationsByKey);
+
+  return Promise.all(
+    entries.map(async ([keyId, values]) => {
+      try {
+        const valueIds = R.map(values, (v) => v.id);
+        const existingId = await findExistingCustomization(keyId, valueIds);
+
+        return existingId ? existingId : createNewCustomization(keyId, valueIds);
+      } catch (error) {
+        console.error("Error creating item customization:", error);
+        throw error;
+      }
+    })
+  );
+};
+
+const findExistingCustomization = async (
```

Comment:
denne er brukt til å ikke lage nye customizations records for hver drink, for eksempel en melk=helmelk

logikken her sluttet å funke når vi tillot flere "melk" som eksempel. Jeg tror ikke det er kostbart i det hele tatt å lage kopier, men det slår meg som unødvendig.

hva tenker du?

Reply:
@IldenH

Reply:
jeg tenker mest på dette i forhold til datanalyse tror jeg. det kan heller være en ide å normalisere data på forkant av eksport

Reply:
Tenkte meg om, fjernes for å være konsistent med resten av kodebasen slik vi gjorde med order_item. Kan heller legge til denne kompleksiteten senere